Telerik.JustMock/Arg.cs

```diff
@@ -23,150 +23,160 @@ JustMock Lite
 
 namespace Telerik.JustMock
 {
-	/// <summary>
-	/// Allows specification of a matching condition for an argument, rather
-	/// a specific value.
-	/// </summary>
-	public static partial class Arg
-	{
-		/// <summary>
-		/// Matches argument for the expected condition.
-		/// </summary>
-		/// <typeparam name="T">
-		/// Contains the type of the argument.
-		/// </typeparam>
-		/// <param name="match">Matcher expression</param>
-		/// <returns>Argument type</returns>
-		[ArgMatcher(Matcher = typeof(PredicateMatcher<>))]
-		public static T Matches<T>(Expression<Predicate<T>> match)
-		{
-			return ProfilerInterceptor.GuardInternal(() =>
-			{
-				MockingContext.CurrentRepository.AddMatcherInContext(new PredicateMatcher<T>(match));
-				return default(T);
-			});
-		}
+    /// <summary>
+    /// Allows specification of a matching condition for an argument, rather
+    /// a specific value.
+    /// </summary>
+    public static partial class Arg
+    {
+        static Arg()
+        {
+            Expr = new ArgExprImplementation();
+        }
 
-		/// <summary>
-		/// Matches argument for the specified range.
+        /// <summary>
+		/// Specifies argument matchers used in non-public method arrangements.
 		/// </summary>
-		/// <typeparam name="T">Type of the argument.</typeparam>
-		/// <param name="from">starting value.</param>
-		/// <param name="to">ending value.</param>
-		/// <param name="kind">Kind of Range</param>
-		/// <returns>Argument type</returns>
-		[ArgMatcher(Matcher = typeof(RangeMatcher<>))]
-		public static T IsInRange<T>(T from, T to, RangeKind kind) where T : IComparable
-		{
-			return ProfilerInterceptor.GuardInternal(() =>
-			{
-				MockingContext.CurrentRepository.AddMatcherInContext(new RangeMatcher<T>(from, to, kind));
-				return default(T);
-			});
-		}
+        public static IArgExpr Expr { get; private set; }
```

Comment:
@mihail-vladov IMO we need to think about guarding this getter with ProfilerInterceptor.GuardInternal.